backend/src/main/java/fk/prof/backend/model/request/CompositeByteBufInputStream.java

```diff
@@ -0,0 +1,110 @@
+package fk.prof.backend.model.request;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.Unpooled;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class CompositeByteBufInputStream extends InputStream {
+  private final CompositeByteBuf buffer;
+  private int readerIndexAtMark = 0;
+
+
+  public CompositeByteBufInputStream() {
+    buffer = Unpooled.compositeBuffer();
+  }
+
+  public CompositeByteBufInputStream(int maxNumComponents) {
+    buffer = Unpooled.compositeBuffer(maxNumComponents);
+  }
+
+  public void accept(ByteBuf newSource) {
+    buffer.addComponent(true, newSource);
+  }
+
+  @Override
+  public int read() throws IOException {
+    if (!buffer.isReadable()) {
+      return -1;
+    }
+    int readByte = buffer.readByte() & 0xff;
+    return readByte;
+  }
+
+  @Override
+  public int read(byte[] b, int off, int len) throws IOException {
+    int available = available();
+    if (available == 0) {
+      return -1;
+    }
+
+    len = Math.min(available, len);
+    buffer.readBytes(b, off, len);
+    return len;
+  }
+
+  @Override
+  public long skip(long n) throws IOException {
+    if (n > Integer.MAX_VALUE) {
```

Comment:
The right thing to do here is to loop skipping Integer.MAX_VALUE until n becomes less than max-value of int. Doesn't matter practically, but one less thing to worry about if done theoretically correctly.

Or just fail it, its easy to test too. Just throw an IllegalArgumentException saying java byte-arrays can't be longer than int max-val.